homeassistant/components/light/yeelight.py

```diff
@@ -46,6 +47,7 @@
     vol.Optional(CONF_TRANSITION, default=DEFAULT_TRANSITION): cv.positive_int,
     vol.Optional(CONF_MODE_MUSIC, default=False): cv.boolean,
     vol.Optional(CONF_SAVE_ON_CHANGE, default=False): cv.boolean,
+    vol.Optional(CONF_MODEL): cv.string,
```

Comment:
Why would we add this? Also doesn't seem related to the PR  description

Reply:
It's not possible to determine the yeelight model via the api. If you provide the model additional features can be enabled by the yeelight library: https://github.com/home-assistant/home-assistant/pull/17305/files#diff-1395165791bf7671502a6e2e7bd7d4cbR286

Reply:
Can't we get the library to just try calling different APIs to see what is available and determine model capabilities like that?

Reply:
In this case we would discover the device twice.

Reply:
I asked about a possibility for getting an extension for the api and the yeelight people were not at least completely against it: https://forum.yeelight.com/t/topic/7075/5 . At the moment that model information is only available either via mDNS or their custom SSDP (related  https://github.com/home-assistant/netdisco/pull/193).

Reply:
Here is an idea: add config entries to Yeelight, add device info so we can store the device model in the registry and read that during setup.